README.md
# animated_radio_buttons

A radio button package for flutter. This widget is useful to quickly add a set of radio buttons to your flutter app.


## Package page 
 
  [animated_radio_buttons](https://pub.dev/packages/animated_radio_buttons)


## Simple Usage

#### Creating a basic `RadioButtons Group`

    
    int myVar ;
    
    AnimatedRadioButtons(
                backgroundColor: Colors.yellow[400],
                value: myVar,
                layoutAxis: Axis.horizontal,
                buttonRadius: 40.0,
                items: [
                  AnimatedRadioButtonItem(
                      label: "Star Battles",
                      color: Colors.black,
                      fillInColor: Colors.grey),
                  AnimatedRadioButtonItem(
                      label: "Star Trek",
                      labelTextStyle: TextStyle(color: Colors.blue),
                      fillInColor: Colors.yellow[400]),
                  AnimatedRadioButtonItem(
                      label: "Losers",
                      color: Colors.green,
                      fillInColor: Colors.greenAccent)
                ],
                onChanged: (value) {
                  setState(() {
                    myVar = value;
                  });
                },
              );

<p>
  <img width="320px" alt="Output" src="https://raw.githubusercontent.com/srihariash999/animated_radio_buttons/master/Screnshots/gif.gif"/>

</p>


CHANGELOG.md
## [0.0.1] - Nov 21, 2020

Initial Release. This widget is useful to quickly add a set of radio buttons to your flutter app

Features:
- RadioButton set with multiple items
- Items can be individually customised.


## [0.0.2] - Nov 21, 2020

Initial Release. This widget is useful to quickly add a set of radio buttons to your flutter app

Features:
- Minor Bug Fix


## [0.0.3] - Nov 21, 2020

Initial Release. This widget is useful to quickly add a set of radio buttons to your flutter app

Features:
- Import Issues fix

## [0.0.4] - Nov 21, 2020

Initial Release. This widget is useful to quickly add a set of radio buttons to your flutter app

Features:
- Import Issues fix


## [0.0.5] - Nov 21, 2020

Initial Release. This widget is useful to quickly add a set of radio buttons to your flutter app

Features:
- Example added.
- ReadMe updated.

## [0.0.6] - Nov 21, 2020

Initial Release. This widget is useful to quickly add a set of radio buttons to your flutter app

Features:
- Example improved.
- ReadMe updated (screenshots).
  
## [0.0.7] - Nov 21, 2020

 This widget is useful to quickly add a set of radio buttons to your flutter app

Features:
- Updated descriptions

## [0.0.8] - Dec 13, 2020

 This widget is useful to quickly add a set of radio buttons to your flutter app

Features:
- Fixed requires hot restart to reflect changes bug
- Can set initial value by setting 'value' variable to an initial value



lib/src/animated_radio_buttons_widget.dart
import 'package:flutter/material.dart';

import 'animated_radio_button_item.dart';

List<double> _radii = [];
double _buttonRadius;

class AnimatedRadioButtons extends StatefulWidget {
  final List<AnimatedRadioButtonItem> items;
  final double buttonRadius;
  final Function(int value) onChanged;

  final Color backgroundColor;
  final Axis layoutAxis;
  final int value;
  final Curve animationCurve;

  AnimatedRadioButtons({
    Key key,
    @required this.items,
    this.buttonRadius,
    @required this.onChanged,
    @required this.value,
    this.layoutAxis,
    this.animationCurve,
    this.backgroundColor,
  }) {
    if (buttonRadius == null) {
      _buttonRadius = 30.0;
    } else {
      _buttonRadius = buttonRadius;
    }

    if (_radii.length == 0) {
      for (int i = 0; i < items.length; i++) {
        if (value != null && i == value) {
          _radii.add(_buttonRadius * 0.20);
        }
        _radii.add(_buttonRadius * 0.70);
      }
    }
  }

  @override
  _AnimatedRadioButtonsState createState() => _AnimatedRadioButtonsState();
}

class _AnimatedRadioButtonsState extends State<AnimatedRadioButtons> {
  updateSelection(int k) {
    setState(() {
      for (int i = 0; i < _radii.length; i++) {
        if (i == k) {
          _radii[i] = _buttonRadius * 0.20;
        } else {
          _radii[i] = _buttonRadius * 0.70;
        }
      }
    });
  }

  provideCallBack(int k) {
    widget.onChanged(k);
  }

  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Material(
      child: Container(
        color: widget.backgroundColor ?? Colors.white,
        alignment: Alignment.center,
        child: Wrap(
          runSpacing: 10.0,
          spacing: 10.0,
          direction: widget.layoutAxis ?? Axis.horizontal,
          alignment: WrapAlignment.spaceEvenly,
          children: widget.items.map(
            (item) {
              int _localIndex = 0;

              for (var i in widget.items) {
                if (i == item) {
                  break;
                } else {
                  _localIndex++;
                }
              }

              return GestureDetector(
                onTap: () {
                  updateSelection(_localIndex);
                  provideCallBack(_localIndex);
                },
                child: Row(
                  mainAxisSize: MainAxisSize.min,
                  children: [
                    Container(
                      alignment: Alignment.center,
                      width: _buttonRadius,
                      height: _buttonRadius,
                      decoration: new BoxDecoration(
                        color: item.getColor ?? Colors.orange,
                        shape: BoxShape.circle,
                      ),
                      child: AnimatedContainer(
                        duration: Duration(milliseconds: 300),
                        curve: widget.animationCurve ?? Curves.bounceIn,
                        width: _radii[_localIndex],
                        height: _radii[_localIndex],
                        decoration: new BoxDecoration(
                          color: item.getFillInColor ?? Colors.white,
                          shape: BoxShape.circle,
                        ),
                      ),
                    ),
                    SizedBox(width: 10.0),
                    Text(
                      item.getLabel,
                      style:
                          item.getLabelTextStyle ?? TextStyle(fontSize: 16.0),
                    ),
                    SizedBox(width: 10.0),
                  ],
                ),
              );
            },
          ).toList(),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:animated_radio_buttons/animated_radio_buttons.dart';

void main() {
  runApp(
    MaterialApp(
      home: Home(),
    ),
  );
}

class Home extends StatefulWidget {
  @override
  _HomeState createState() => _HomeState();
}

class _HomeState extends State<Home> {
  int myVar = 3;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.yellow[400],
      appBar: AppBar(),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
        children: [
          AnimatedRadioButtons(
            backgroundColor: Colors.yellow[400],
            value: myVar,
            layoutAxis: Axis.horizontal,
            buttonRadius: 40.0,
            items: [
              AnimatedRadioButtonItem(
                  label: "Star Battles",
                  color: Colors.black,
                  fillInColor: Colors.grey),
              AnimatedRadioButtonItem(
                  label: "Star Trek",
                  labelTextStyle: TextStyle(color: Colors.blue),
                  fillInColor: Colors.yellow[400]),
              AnimatedRadioButtonItem(
                  label: "Losers",
                  color: Colors.green,
                  fillInColor: Colors.greenAccent)
            ],
            onChanged: (value) {
              setState(() {
                myVar = value;
              });
            },
          ),
          myVar == null
              ? Text(" Current selected index: *please select one* ")
              : Text(" Current selected index: $myVar")
        ],
      ),
    );
  }
}
